api/v1alpha1/gateway_types.go

```diff
@@ -64,22 +61,56 @@ type GatewaySpec struct {
 	Routes []core.TypedLocalObjectReference `json:"routes"`
 }
 
+const (
+	// HTTPProcotol constant.
+	HTTPProcotol = "HTTP"
+	// HTTPSProcotol constant.
+	HTTPSProcotol = "HTTPS"
+)
+
 // Listener defines a
 type Listener struct {
 	// Address bound on the listener. This is optional and behavior
 	// can depend on GatewayClass. If a value is set in the spec and
 	// the request address is invalid, the GatewayClass MUST indicate
 	// this in the associated entry in GatewayStatus.Listeners.
 	//
+	// Support:
+	//
 	// +optional
 	Address *ListenerAddress `json:"address"`
-	// Ports is a list of ports associated with the Address.
-	Ports []ListenerPort `json:"ports"`
+	// Port is a list of ports associated with the Address.
+	//
+	// Support:
+	// +optional
+	Port *int `json:"port"`
+	// Protocol to use.
+	//
+	// Support:
+	// +optional
+	Protocol *string `json:"protocol"`
+	// TLS configuraton for the Listener.
+	//
+	// Support:
+	// +optional
+	TLS *ListenerTLS `json:"tls"`
+	// Extension for this Listener.
+	//
+	// Support: custom.
+	// +optional
+	Extension *core.TypedLocalObjectReference `json:"extension"`
 }
 
 const (
-	IPAddressType    = "IPAddress"
-	NamedAddressType = "NamedAddress"
+	// IPAddress is an address that is an IP address.
+	//
+	// Support: Extended.
+	IPAddress = "IPAddress"
+	// NamedAddress is an address selected by name. The interpretation of
+	// the name is depenedent on the controller.
```

Comment:
I am a bit confused what this would be. A hostname like `foo.com`?

Reply:
Idea is more like `us-central1/my-ip-address`
We are going to go over the sketch + API types in detail, the current status of the types is to mostly just have some CRDs to test out the kubebuilder infrastructure/test/CI infrastructure.

Reply:
Keep your comment in mind when we go over the gateway APIs :-)